---
title: "sponge_tank"
author: "cai wang"
date: "14/10/2021"
output: html_document
---

```{r setup, include=FALSE}
library(tidyverse)
library(phyloseq)
library("ggplot2")
library(decontam)
library(data.table)
library(sjmisc)
library(conflicted)
conflict_prefer("filter", "dplyr")
conflict_prefer("select", "dplyr")
```
#######otu collapsing##########
```{r separate sequnce}
OTUseq<-read.csv("spongetank_all_SWARM3_FINAL_OTU.csv", header=TRUE)
seq<-apply(OTUseq, 1, function(x){paste0(">",x[1],"\n",x[121])})
writeLines(seq)
write.table(seq,file="spongetank_all_SWARM3_FINAL_OTU.fasta",row.names = F,quote=F)
#delete x
names(OTUseq)
OTUtab<-OTUseq %>% select (D1.SW.BLANK:LIB02.PCR.Pos.03)
rownames(OTUtab)=OTUseq$id
```

```{r code for LuLu (use to fiter ech OTU), eval=FALSE }
#1.#filter echo OTU
##terminal##vsearch --usearch_global spongetank_all_SWARM3_FINAL_OTU.fasta --db spongetank_all_SWARM3_FINAL_OTU.fasta --self --id .84 --iddef 1 --userout match_list.txt -userfields query+target+id --maxaccepts 0 --query_cov .9 --maxhits 10 
#install.packages("devtools")
#library(devtools)
#install_github("tobiasgf/lulu") 
library(lulu)
matchlist <- read.table("match_list.txt", header=FALSE,as.is=TRUE, stringsAsFactors=FALSE)

curated_result <- lulu(OTUtab, matchlist, minimum_ratio_type = "min", minimum_ratio = 1, minimum_match = 84, minimum_relative_cooccurence = 0.95)
#compare the result
#curated_result$curated_table
#curated_result$original_table
curated_result$discarded_coun
OTUtax <-OTUseq[OTUseq$id %in% rownames(curated_result$curated_table),]
write.csv(OTUtax, file = "spongetank_SWARM3_OTU_LULU_FULL_20211019.csv",row.names = TRUE, quote = FALSE)
#write.csv(curated_result$curated_table, file = "spongetank_SWARM_OTU_LULU_20211014.csv",row.names = TRUE, quote = FALSE)
rm(list=ls())
```

```{r collapse OTUs on terminal}
#collapse MOTUs on terminal
#Rscript /Users/wang/Desktop/stefano_lab/bioinformatics/ob1_mbc/R_scripts_metabarpark/owi_collapse -s 16 -e 116 -t 0.50 -i spongetank_SWARM5_OTU_LULU_FULL_20211019.csv
#-s 14 Sample columns start; -e sample columns end. Default = 98; -t 0.50 Threshold for collapsing
```

#######clean up：remove low count and small OTUs##########
```{r phyloseq formal}
OTUseq<-read.csv("spongetank_SWARM3_OTU_LULU_FULL_20211019_collapsed.csv", header=TRUE)
#otu_table
OTU<-OTUseq %>% select (D1.SW.BLANK:LIB02.PCR.Pos.03)
rownames(OTU)=OTUseq$id

#tax_table
TAX<-OTUseq %>% select (kingdom_name:species_name)
rownames(TAX)=OTUseq$id
TAX <- as.matrix(TAX)

#sample_data
sample_data<-read.csv("sampledata20211014.csv", header=TRUE)
sample_data=arrange(sample_data,name)
sample_data$quant_reading=colSums(OTU)
rownames(sample_data)=sample_data$name

#####filter total reads=0 samples
sample_data=sample_data %>% filter(quant_reading!=0)
OTU= OTU %>% select(sample_data$name)
OTU <- as.matrix(OTU)
###write the phyloseq format
physeq_spongetank = phyloseq(otu_table(OTU, taxa_are_rows = TRUE), tax_table(TAX), sample_data(sample_data))

test=data.frame(otu_table(physeq_spongetank))
colSums(test)
test$D1.TANKB.WATER
```

```{r decontam}
contamdf.freq <- isContaminant(physeq_spongetank, method="frequency", conc="quant_reading",batch = "RUN")
table(contamdf.freq$contaminant)
decontam.T=contamdf.freq %>% filter(contaminant=="TRUE")
#plot all the decontam otus
T.n=which(rownames(OTU)%in% rownames(decontam.T)==TRUE)
plot_frequency(physeq_spongetank, taxa_names(physeq_spongetank)[c(T.n)], conc="quant_reading") + 
  xlab("DNA Concentration (PicoGreen fluorescent intensity)")
TAX[T.n,7]
#delect decontam=Ture seq.
ps.noncontam <- prune_taxa(!contamdf.freq$contaminant, physeq_spongetank)

save(ps.noncontam,physeq_spongetank,file = "physeq_spongetank_noncontam.rdata")
```

we know that small OTUs are morelikely to be false OTUs. But what is small?  One possible method is that we use a heuristic method from this phyloseq tutorial:
http://evomics.org/wp-content/uploads/2016/01/phyloseq-Lab-01-Answers.html

```{r phyloseq, eval=FALSE, include=FALSE}
load("physeq_spongetank_noncontam.rdata")
OTU<- data.frame(t(as.data.frame(otu_table(ps.noncontam))))
metadata<- data.frame(sample_data(ps.noncontam)) 
#because there are no fiter steps basing on count, so the table contains all the sequence. 
li3<- OTU %>% filter (metadata$RUN==3)
li4<- OTU %>% filter (metadata$RUN==4)
li3[li3<10] <- 0#i choose 10 because there are 11 reads in a positive sample, but there are 80 reads in a negtive sample, so it's not the best, just for now
li4[li4<20] <- 0#i choose 20 because there are 20 reads in a pos samples that isn't the pos species.
#i didn't remove otus according to single read counts in each control.
colSums(li3)
colSums(li4)
OTU=bind_rows(li3,li4)
rm(li3,li4)
########## filter out small OTUs, phyloseq ##
TotalCounts <- c(colSums(OTU)) 
TotalCounts # number of reads per OTU ("Cluster") (summed over all samples)
tdt = data.table(colnames(OTU), TotalCounts = colSums(OTU), OTU = colnames(OTU)) # data.table() package

# histogram of OTU sizes.  You can 
ggplot(tdt, aes(TotalCounts)) + 
  geom_histogram() + 
  ggtitle("Histogram of Total Counts")

tdt[(TotalCounts <= 0), .N] # 42 OTUs with 0 reads
tdt[(TotalCounts <= 1), .N] # 42 OTUs with 1 read (single-read OTUs)
tdt[(TotalCounts <= 2), .N] # 42 OTUs with <=2 reads (two-read OTUs)

# these are data.table commands, and I am only following the phyloseq tutorial
taxcumsum = tdt[, .N, by = TotalCounts]
setkey(taxcumsum, TotalCounts)
taxcumsum[, CumSum := cumsum(N)]
# Define the plot
pCumSum = ggplot(taxcumsum, aes(TotalCounts, CumSum)) + 
  geom_point() +
  xlab("Filtering Threshold, Minimum Total Counts") +
  ylab("OTUs Filtered") +
  ggtitle("OTUs that would be filtered vs. the minimum count threshold")
pCumSum # this is the ggplot2 object
pCumSum + xlim(0, 200)
pCumSum + xlim(0, 50)

# after viewing the above figure, we chose 50 as the minimum OTU size
# each column is an OTU. We sum the reads per OTU and only keep the columns (OTUs) that have >= 50 reads
commMBC.50 <- OTU[ ,colSums(OTU)>=50] # 53 otus remain
rowSums(commMBC.50) # the total number of reads (summed over all OTUs) per sample (row)
commMBC.50 <- commMBC.50[rowSums(commMBC.50)>0, ] # for safety, remove any samples that have no OTUs (2 samples removed)
rowSums(commMBC.50)
commMBC.50 <- rotate_df(commMBC.50) # transpose to new table (OTUs in rows, samples in columns)

# save the new reduced dataset
metadata
new=which(metadata$name %in% colnames(commMBC.50)==TRUE)
TAX=data.frame(tax_table(ps.noncontam))
metadata=metadata[new,]
new2=which(rownames(TAX)%in% rownames(commMBC.50)==TRUE)
TAX=TAX[new2,]
OTU_TAX=bind_cols(TAX,commMBC.50)

write.csv(OTU_TAX, file = "OTU_TAX_20211019.csv",row.names = TRUE, quote = FALSE)
commMBC.50=as.matrix(commMBC.50)
TAX=as.matrix(TAX)
physeq_spongetank = phyloseq(otu_table(commMBC.50, taxa_are_rows = TRUE), tax_table(TAX), sample_data(metadata))

save(physeq_spongetank,file = "physeq_spongetank_noncontam_clean_20211019.rdata")
```

#######statistical analysis start here##########
```{r tax_name correction}
rm(list=ls())
load("physeq_spongetank_noncontam_clean_20211019.rdata")
OTU<- data.frame(t(as.data.frame(otu_table(physeq_spongetank))))
metadata<- data.frame(sample_data(physeq_spongetank)) 
OTUname=data.frame(OTU=rownames(TAX),family=TAX$family_name,genus=TAX$genus_name,species=TAX$species_name)
OTUname$species=as.character(OTUname$species)
OTUname$OTU=as.character(OTUname$OTU)
OTUname=raster::trim(OTUname)

for (i in 1:length(OTUname$species)) {
  if(OTUname$species[i]=="")
    OTUname$species[i] = paste0(OTUname$genus[i]," ","sp" )
  if(OTUname$genus[i]=="")
    OTUname$species[i] = paste0("f_",OTUname$family[i],"sp" )
}
OTUname$species=gsub(" ", "_", OTUname$species)

c1=which(OTUname$species=="Amphiprion_sp")[1]
c2=which(OTUname$species=="Amphiprion_sp")[2]
OTUname$species[c1]="Amphiprion_sp1"
OTUname$species[c2]="Amphiprion_sp2"
for (i in 1:length(OTUname$species)) {
  if(OTUname$species[i]=="f_sp")
    OTUname$species[i] = OTUname$OTU[i] 
}

TAX$species_name=OTUname$species
#collapse two Zebrasoma sp into one otu
for (i in 1:length(OTU)) {
  if(colnames(OTU)[i]==OTUname$OTU[i])
    colnames(OTU)[i]=OTUname$species[i]
}
OTU <- OTU %>% mutate(Zebrasoma = `Zebrasoma_desjardinii` + Zebrasoma_flavescens) 
OTU <- OTU %>% select(-Zebrasoma_desjardinii,-Zebrasoma_flavescens)
OTU <- rotate_df(OTU)
OTU=as.matrix(OTU)
TAX=bind_rows(TAX,TAX[which(TAX$species_name=="Zebrasoma_desjardinii"),])
TAX$species_name[54]="Zebrasoma"
rownames(TAX)=TAX$species_name
TAX=TAX %>% filter(species_name!="Zebrasoma_desjardinii" & species_name!="Zebrasoma_flavescens")
TAX$genus_name=as.character(TAX$genus_name)
TAX$family_name=as.character(TAX$family_name)
TAX=raster::trim(TAX)
for (i in 1:length(TAX$genus_name)) {
  if(TAX$genus_name[i]=="")
    TAX$genus_name[i] = paste0("unassigned")
  if(TAX$family_name[i]=="")
    TAX$family_name[i] = paste0("unassigned")
}
TAX=as.matrix(TAX)

physeq_spongetank = phyloseq(otu_table(OTU, taxa_are_rows = TRUE), tax_table(TAX), sample_data(metadata))
save(physeq_spongetank,file = "physeq_spongetank_noncontam_clean_20211019.rdata")
rm(list=ls())

#add no sequence sample, and correct tank_000000035 as Gramma_loreto
load("physeq_spongetank_noncontam_clean_20211019.rdata")
sample_date_full<-read.csv("sampledata_full_20211020.csv", header=TRUE)
OTU<- data.frame(otu_table(physeq_spongetank))
#metadata<- data.frame(sample_data(physeq_spongetank)) 
rownames(OTU)[which(rownames(OTU)=="tank_000000035")]=rownames(OTU)[which(rownames(OTU)=="tank_000000035")]="Gramma_loreto"
new=which(sample_date_full$name%in% colnames(OTU)==F)
sample_date_full$name=as.character(sample_date_full$name)
newname=sample_date_full$name[which(sample_date_full$name%in% colnames(OTU)==F)]

OTU.full=matrix(data = "0", nrow = length(OTU$D2.FILT.BLANK), ncol = length(newname), byrow = FALSE,dimnames = NULL)
OTU.full=data.frame(apply(OTU.full, 2, as.numeric))
rownames(OTU.full)=rownames(OTU)
colnames(OTU.full)=newname
OTU.full2=bind_cols(OTU,OTU.full)
TAX=data.frame(tax_table(physeq_spongetank))
rownames(TAX)[which(rownames(TAX)=="tank_000000035")]=rownames(TAX)[which(rownames(TAX)=="tank_000000035")]="Gramma_loreto"
sample_date_full=arrange(sample_date_full,name)
or<-order(colnames(OTU.full2))
OTU.full2<-OTU.full2[,or]
rownames(sample_date_full)=sample_date_full$name
physeq_spongetank = phyloseq(otu_table(as.matrix(OTU.full2), taxa_are_rows = TRUE), tax_table(as.matrix(TAX)), sample_data(sample_date_full))
save(physeq_spongetank,file = "physeq_spongetank_noncontam_clean_all_20211020.rdata")
rm(list=ls())
```

```{r species read abundance}
color=c(RColorBrewer::brewer.pal(8,"Set3"),RColorBrewer::brewer.pal(8,"Set1"),"darkgoldenrod4","#00008B",RColorBrewer::brewer.pal(8,"Set2"))

load("physeq_spongetank_noncontam_clean_all_20211020.rdata")
#add a factor to separate all control samples
sample <- get_variable(physeq_spongetank, "sponge_type") %in% c("SP.CREAM","SP.PINK","SP.BLACK","WATER")
sample_data(physeq_spongetank)$sample <- as.character(sample)
sponge_type2 <- get_variable(physeq_spongetank, "sponge_type") %in% c("SP.CREAM","SP.PINK","SP.BLACK")
sample_data(physeq_spongetank)$sponge_type2 <- as.character(sponge_type2)
physeq_spongetank <- prune_taxa(taxa_sums(physeq_spongetank) > 0, physeq_spongetank)

physeq.tank.true <- subset_samples(physeq_spongetank, experiment!="reef" & sample=="TRUE")
physeq.tank.false <- subset_samples(physeq_spongetank, experiment!="reef" & sample=="FALSE")
physeq.reef <- subset_samples(physeq_spongetank, experiment=="reef")

#figure: entire tank sample with raw reads
#facet_grid=~tank
#geom_point(aes(x=sponge_type, y=Abundance), color="black", position="jitter", size=1)+scale_fill_manual(values=color)
plot_bar(physeq.tank.true,fill="genus_name",facet_grid=~sponge_type2)+scale_fill_manual(values=color)+ geom_bar(aes(fill=genus_name), stat="identity", position="stack")+labs(title="tank samples overview",x="sample",y="raw reads")+theme(axis.text.x = element_text(size=6))#+scale_x_discrete(limits=names)

plot_bar(physeq.tank.false,fill="genus_name")+scale_fill_manual(values=color)+ geom_bar(aes(fill=genus_name), stat="identity", position="stack")+labs(title="tank control overview",x="sample",y="raw reads")+theme(axis.text.x = element_text(size=6))

plot_bar(physeq.reef, fill="genus_name",facet_grid=~sponge_type)+scale_fill_manual(values=color)+ geom_bar(aes(fill=genus_name), stat="identity", position="stack")+labs(title="reef overview",x="sample",y="raw reads")#+scale_x_discrete(limits=c("Fish","Amphibian","Waterfowl","Mammal","other Bird","Domestic"))
#heatmap of control
physeq_spongetank_sno0 <- prune_samples(sample_sums(physeq_spongetank)>0, physeq_spongetank)
physeq.tank.false_sno0 <- subset_samples(physeq_spongetank_sno0, experiment!="reef" & sample=="FALSE")
plot_heatmap(physeq.tank_sno0.false, "NMDS", "bray", "name", "genus_name", low="#66CCFF", high="#000033", na.value="white")

```
#########from now on, reads be converted to read proportion
```{r species read proportion}
rm(list=ls())
color=c(RColorBrewer::brewer.pal(8,"Set3"),RColorBrewer::brewer.pal(8,"Set1"),"darkgoldenrod4","#00008B",RColorBrewer::brewer.pal(8,"Set2"))

load("physeq_spongetank_noncontam_clean_all_20211020.rdata")
OTU<- data.frame(otu_table(physeq_spongetank))
rownames=rownames(OTU)
OTU<- apply(OTU,2,as.numeric)
OTU.proportion=data.frame(apply(OTU, 2, function(x) x/sum(x)))
rownames(OTU.proportion)=rownames
OTU.proportion[is.na(OTU.proportion)] <- 0

physeq_pro = phyloseq(otu_table(as.matrix(OTU.proportion), taxa_are_rows = TRUE), tax_table(physeq_spongetank), sample_data(physeq_spongetank))
par(mar = c(10, 4, 4, 2) + 0.1) # make more room on bottom margin
N <- 30
barplot(sort(taxa_sums(physeq_pro), TRUE)[1:N]/nsamples(physeq_pro), las=2,ylim = c(0,0.2),cex.names = 0.7)

physeq_pro_sno0 <- prune_samples(sample_sums(physeq_pro)>0, physeq_pro)
plot_heatmap(physeq_pro_sno0, "NMDS", "bray", "name", "genus_name", low="#66CCFF", high="#000033", na.value="white")
sponge <- get_variable(physeq_pro, "sponge_type") %in% c("SP.CREAM","SP.PINK","SP.BLACK","WATER")
sample_data(physeq_pro)$sponge <- as.character(sponge)
sponge_type2 <- get_variable(physeq_pro, "sponge_type") %in% c("SP.CREAM","SP.PINK","SP.BLACK")
sample_data(physeq_pro)$sponge_type2 <- as.character(sponge_type2)
physeq_pro <- prune_taxa(taxa_sums(physeq_pro) > 0, physeq_pro)

physeq.tank.true <- subset_samples(physeq_pro, experiment!="reef" & sponge=="TRUE")
physeq.tank.true.sponge <- subset_samples(physeq_pro, experiment!="reef" & sponge2=="TRUE")
physeq.tank.true.water <- subset_samples(physeq.tank.true,sponge2=="FALSE" )
physeq.tank.false <- subset_samples(physeq_pro, experiment!="reef" & sponge=="FALSE")

#figure: entire tank sample with raw reads
#facet_grid=~tank
#geom_point(aes(x=sponge_type, y=Abundance), color="black", position="jitter", size=1)+scale_fill_manual(values=color)
plot_bar(physeq.tank.true,fill="genus_name",facet_grid=~sponge2)+scale_fill_manual(values=color)+ geom_bar(aes(fill=genus_name), stat="identity", position="fill")+labs(title="tank samples overview",x="sample",y="reads proportion")+theme(axis.text.x = element_text(size=6))

plot_bar(physeq.tank.false,fill="genus_name")+scale_fill_manual(values=color)+ geom_bar(aes(fill=genus_name), stat="identity", position="fill")+labs(title="tank samples overview",x="sample",y="reads proportion")+theme(axis.text.x = element_text(size=6))

#计算数据集中每个OTU的累积分数丰度。在这个条形图中，我们通过样本总数（280）进一步标准化
par(mar = c(10, 4, 4, 2) + 0.1) # make more room on bottom margin
N <- 20
barplot(sort(taxa_sums(physeq.tank.true.sponge), TRUE)[1:N]/nsamples(physeq.tank.true.sponge), las=2,cex.names = 0.7,ylim = c(0,0.2))
barplot(sort(taxa_sums(physeq.tank.true.water), TRUE)[1:N]/nsamples(physeq.tank.true.water), las=2,cex.names = 0.7,ylim = c(0,0.3))

```

```{r time decline}
metadata<- data.frame(sample_data(physeq.tank.true))
metadata$hours_fish=as.numeric(as.character(metadata$hours_fish))
metadata$hours_fish=metadata$hours_fish*(-1)
OTU<-rotate_df(data.frame(otu_table(physeq.tank.true)))
OTU$name=rownames(OTU)
metadata.hour=left_join(metadata,OTU,by="name")

metadata.hour.Amphiprion=metadata.hour%>% mutate(species="Amphiprion")%>% rename(read = Amphiprion_sp1)
metadata.hour.Chromis= metadata.hour%>% mutate(species="Chromis")%>% rename(read = Chromis_viridis)
metadata.hour.Zebrasom= metadata.hour%>% mutate(species="Zebrasoma")%>% rename(read = Zebrasoma)
metadata.hour.Gramma= metadata.hour%>% mutate(species="Gramma")%>% rename(read = Gramma_loreto)
metadata.hour.Hippocampus= metadata.hour%>% mutate(species="Hippocampus")%>% rename(read = Hippocampus_kuda)

metadata.hour.new=bind_rows(metadata.hour.Amphiprion,metadata.hour.Chromis,metadata.hour.Zebrasom,metadata.hour.Gramma,metadata.hour.Hippocampus)
rm(metadata.hour.Amphiprion,metadata.hour.Chromis,metadata.hour.Zebrasom,metadata.hour.Gramma,metadata.hour.Hippocampus)

metadata.hour.new2=metadata.hour.new%>%select(name:sponge_type,species,read)

metadata.hour.new2$read=as.numeric(as.character(metadata.hour.new2$read))
#metadata.hour.new2$read=log10(metadata.hour.new2$read)
#ggplot(data=metadata.hour.new2, aes(x=hours_fish, y=read, colour= species))+ geom_line(size=0.3)+geom_point()+facet_grid(sponge_type ~ tank)+labs(title="tank time decline",x="hours",y="read proportion")

metadata.hour.new2$read2=log2(metadata.hour.new2$read)
#metadata.hour.new2$sponge_type2=paste0(metadata.hour.new2$sponge_type,metadata.hour.new2$tank)


metadata.hour.new2 <- metadata.hour.new2 %>%
  mutate(
    species = case_when(
      species == "Amphiprion" ~ "1",
      species == "Chromis" ~ "4",
      species == "Zebrasoma" ~ "7",
      species == "Gramma" ~ "10",
      species == "Hippocampus" ~ "13",
      TRUE ~ as.character(species)
    )
  )
metadata.hour.new2$species=as.numeric(metadata.hour.new2$species)
for (i in 1:length(metadata.hour.new2$name)) {
  if(metadata.hour.new2$sponge_type[i]=="SP.CREAM")
  metadata.hour.new2$species[i]=metadata.hour.new2$species[i]+0.7
  if(metadata.hour.new2$sponge_type[i]=="SP.PINK")
  metadata.hour.new2$species[i]=metadata.hour.new2$species[i]+1.4
}

ggplot(data = metadata.hour.new2,
       aes(x = as.factor(hours_fish), y = species, size = read2,col=sponge_type)) +scale_y_continuous(breaks = c(1, 4, 8, 12, 16))+geom_point() + theme_bw() +scale_color_manual(values=c("#8B7D6B", "#FFB90F", "#8B008B","#7AC5CD"))+theme(panel.grid.major=element_line(colour=NA))+scale_size("log2(read count)")+facet_grid(~tank)#+scale_y_discrete(limits=c("Amphiprion","Chromis","Zebrasoma","Gramma","Hippocampus"))#+facet_grid(sponge_type~ tank)
```

```{r incidence}
OTU.target=OTU%>%select(Amphiprion_sp1,Chromis_viridis,Zebrasoma,Gramma_loreto,Hippocampus_kuda)
OTU.target[OTU.target > 0] <- 1
OTU.target$incidence=rowSums(OTU.target)
OTU.target$name=rownames(OTU.target)
metadata.hour.new3=left_join(metadata,OTU.target,by="name")
str(metadata.hour.new3)

ggplot(metadata.hour.new3, aes(x=factor(hours_fish), y=incidence,colour=sponge_type)) + 
  geom_boxplot() +
  labs(title = "incidence for target spcies only")+
  geom_jitter(width =0.2,size=1,height=0.3)+ 
  theme_bw()+ylab("incidence")+xlab("time")+ylab("incidence")+xlab("time")+scale_color_manual(values=c("#8B7D6B", "#FFB90F", "#8B008B","#7AC5CD"))

ggplot(metadata.hour.new3, aes(x = hours_fish, y = incidence, colour=sponge_type)) +geom_jitter(aes(colour=sponge_type,shape=tank),height=0.3,width =0.6,size=1.5)+geom_smooth(method = "glm",se=F)+scale_x_continuous( breaks=seq(-22,72,4))+ylab("incidence")+xlab("time")+scale_color_manual(values=c("#8B7D6B", "#FFB90F", "#8B008B","#7AC5CD"))

#GLM
metadata.hour.new3.PINK=metadata.hour.new3%>%filter(sponge_type=="SP.PINK")
lm.pink<-lm(metadata.hour.new3.PINK$hours_fish~metadata.hour.new3.PINK$incidence)
summary(lm.pink)# p-value =0.1506,0.05853
metadata.hour.new3.WATER=metadata.hour.new3%>%filter(sponge_type=="WATER")
lm.WATER<-lm(metadata.hour.new3.WATER$hours_fish~metadata.hour.new3.WATER$incidence)
summary(lm.WATER)
metadata.hour.new3.CREAM=metadata.hour.new3%>%filter(sponge_type=="SP.CREAM")
lm.CREAM<-lm(metadata.hour.new3.CREAM$hours_fish~metadata.hour.new3.CREAM$incidence)
summary(lm.CREAM)

#using cor.test to test correlation
#cor.test(y=habitatN$infestation_rate, x = habitatN$lvs2, method = "pearson")

#metadata.hour.new3$sponge_type=as.character(metadata.hour.new3$sponge_type)
#ggplot(data=metadata.hour.new3, aes(x=hours_fish, y=incidence, colour= sponge_type))+ geom_line(size=0.3)+geom_point()+facet_grid(~tank)+labs(title="tank time decline",x="hours",y="read proportion")

```

```{r}
TukeyHSD(aov(metadata.hour.new3$incidence~metadata.hour.new3$sponge_type)) 
plot(TukeyHSD(aov(metadata.hour.new3$incidence~metadata.hour.new3$sponge_type)))
library(ggsignif)
library(gtools)
#compared_list
compared_list=list()

levels=levels(as.factor(metadata.hour.new3$sponge_type))
combinations=combinations(4,2,levels)
for (i in 1:(length(combinations)/2)) {
  compared_list[[i]]=combinations[i,]
}
ggplot(metadata.hour.new3, aes(x=sponge_type, y=incidence)) + 
  geom_boxplot() +
  labs(title = "incidence t.test")+
  geom_signif(comparisons = compared_list, test = t.test, step_increase = 0.2,map_signif_level=TRUE)+geom_jitter(width =0.3,size=1,height=0.15)
```

