---
title: "spongeDNA_ob1_code"
author: "cai wang"
date: "20/12/2021"
output: html_document
---
```{r setup, include=FALSE}
library(tidyverse)
library(phyloseq)
library(ggplot2)
library(data.table)
library(sjmisc)
library(vegan)
library(pheatmap)
library(RColorBrewer)
library(car)
library(ggrepel)
library(mvabund)
library(HH)
library(conflicted)
conflict_prefer("filter", "dplyr")
conflict_prefer("select", "dplyr")
```
#######first filter: otu collapsing##########
firstly, extracting the sequences from the final file 
```{r separate sequnce}
OTUseq<-read.csv("~/src/sponge_DNA/spongDNA_ob1/sep_reef/spongetank_10reads_SWARM3_FINAL_OTUs.csv", header=TRUE)
seq<-apply(OTUseq, 1, function(x){paste0(">",x[1],"\n",x[109])})
writeLines(seq)
write.table(seq,file="spongetank_10reads_SWARM3_FINAL_OTUs.fasta",row.names = F,quote=F)
#delete x
names(OTUseq)
OTUtab<-OTUseq %>% select (D1.SW.BLANK:LIB02.PCR.Pos.03)
rownames(OTUtab)=OTUseq$id
```
then using lulu to collapse
```{r code for LuLu (use to fiter ech OTU), eval=FALSE }
#1.#filter echo OTU
##terminal##vsearch --usearch_global spongetank_10reads_SWARM3_FINAL_OTUs.fasta --db spongetank_10reads_SWARM3_FINAL_OTUs.fasta --self --id .84 --iddef 1 --userout match_list.txt -userfields query+target+id --maxaccepts 0 --query_cov .9 --maxhits 10 
#install.packages("devtools")
#library(devtools)
#install_github("tobiasgf/lulu") 
library(lulu)
matchlist <- read.table("match_list.txt", header=FALSE,as.is=TRUE, stringsAsFactors=FALSE)

curated_result <- lulu(OTUtab, matchlist, minimum_ratio_type = "min", minimum_ratio = 1, minimum_match = 84, minimum_relative_cooccurence = 0.95)
#compare the result
#curated_result$curated_table
#curated_result$original_table
curated_result$discarded_coun
OTUtax <-OTUseq[OTUseq$id %in% rownames(curated_result$curated_table),]
write.csv(OTUtax, file = "spongetank_10reads_SWARM3_OTU_LULU_20211207.csv",row.names = F, quote = FALSE)
rm(list=ls())
```
after using lulu, there still some OTUs obviously belong to same species like homo, so use metabarpark script to collapse again. 
```{r collapse OTUs on terminal}
#collapse MOTUs on terminal
#Rscript /Users/wang/Desktop/stefano_lab/bioinformatics/ob1_mbc/R_scripts_metabarpark/owi_collapse -s 16 -e 116 -t 0.50 -i spongetank_SWARM5_OTU_LULU_FULL_20211019.csv
#-s 14 Sample columns start; -e sample columns end. Default = 98; -t 0.50 Threshold for collapsing
```
#######second filter: filtering OTU with reads <10
according to positive control, The smallest read is 11, so we select 10 as the threshold to filter "OTU[OTU <= 10] <- 0"and we plot all control samples with unfiltered OTU and reads.#####

before filter we do two more formative steps:
one: transfer data to phyloseq format
```{r phyloseq formal}
OTUseq<-read.csv("spongetank_10reads_SWARM3_OTU_LULU_20211207_collapsed_manual.csv", header=TRUE)
#otu_table
names(OTUseq)
OTU<-OTUseq %>% select (D1.SW.BLANK:LIB02.PCR.Pos.03)
rownames(OTU)=OTUseq$id

#tax_table
TAX<-OTUseq %>% select (kingdom_name:species_name)
rownames(TAX)=OTUseq$id
TAX <- as.matrix(TAX)

#sample_data
sample_data<-read.csv("sampledata_full_20211207.csv", header=TRUE)
sample_data=arrange(sample_data,name)
otusum=data.frame(name=names(OTU),quant_reading=colSums(OTU))
otusum=arrange(otusum,name)
sample_data=left_join(sample_data,otusum,by="name")
rownames(sample_data)=sample_data$name

#####filter total reads=0 samples
sample_data=sample_data %>% filter(quant_reading!=0)
OTU= OTU %>% select(sample_data$name)
OTU <- as.matrix(OTU)
###write the phyloseq format
physeq_spongetank = phyloseq(otu_table(OTU, taxa_are_rows = TRUE), tax_table(TAX), sample_data(sample_data))

test=data.frame(otu_table(physeq_spongetank))
colSums(test)
save(physeq_spongetank,file = "physeq_spongetank.rdata")
```
two: we correct species name manually, each OTU assigned to a species will be renamed to its species name
```{r tax_name correction}
OTU<- data.frame(t(as.data.frame(otu_table(physeq_spongetank))))
metadata<- data.frame(sample_data(physeq_spongetank)) 
TAX=data.frame(tax_table(physeq_spongetank))
OTUname=data.frame(OTU=rownames(TAX),family=TAX$family_name,genus=TAX$genus_name,species=TAX$species_name)
OTUname$species=as.character(OTUname$species)
OTUname$OTU=as.character(OTUname$OTU)
OTUname=raster::trim(OTUname)

for (i in 1:length(OTUname$species)) {
  if(OTUname$species[i]=="")
    OTUname$species[i] = paste0(OTUname$genus[i]," ","sp" )
  if(OTUname$genus[i]=="")
    OTUname$species[i] = paste0("f_",OTUname$family[i],"sp" )
}
for (i in 1:length(OTUname$species)) {
  if(OTUname$species[i]=="f_sp")
    OTUname$species[i] = paste0("OTU",i)
}
OTUname$species=gsub(" ", "_", OTUname$species)


TAX$species_name=OTUname$species
#collapse two Chromis and Zebrasoma sp into one otu
names(OTU)
OTU <- OTU %>% mutate(Chromis_viridis = 	tank_000000068 + tank_000000005) 
OTU <- OTU %>% select(-tank_000000068,-tank_000000005)
OTU <- OTU %>% mutate(Zebrasoma_veliferum = tank_000000388 + tank_000000008) 
OTU <- OTU %>% select(-tank_000000388,-tank_000000008)
OTU <- rotate_df(OTU)
OTU$OTU=rownames(OTU)
OTU2=left_join(OTU,OTUname,by="OTU")
OTU2$species[54]=OTU2$OTU[54]
OTU2$species[55]=OTU2$OTU[55]
rownames(OTU2)=OTU2$species
rownamesOTU2=rownames(OTU2)
OTU2 <- OTU2 %>% select(-OTU,-family,-species,-genus)
OTU2<- apply(OTU2,2,as.numeric)
rownames(OTU2)=rownamesOTU2

TAX$OTU=rownames(TAX)
TAX <- TAX %>% filter(OTU!="tank_000000068" & OTU!="tank_000000388")
rownames(TAX)=TAX$species_name
TAX <- TAX %>% select(-OTU)
TAX$genus_name=as.character(TAX$genus_name)
TAX$family_name=as.character(TAX$family_name)
TAX=raster::trim(TAX)
for (i in 1:length(TAX$genus_name)) {
  if(TAX$genus_name[i]=="")
    TAX$genus_name[i] = paste0("unassigned")
  if(TAX$family_name[i]=="")
    TAX$family_name[i] = paste0("unassigned")
}
for (i in 1:length(TAX$species_name)) {
  if(TAX$genus_name[i]=="unassigned")
    TAX$species_name[i]=paste0("unassigned")
}
TAX=as.matrix(TAX)

physeq_spongetank = phyloseq(otu_table(OTU2, taxa_are_rows = TRUE), tax_table(TAX), sample_data(metadata))
save(physeq_spongetank,file = "physeq_spongetank_20211220.rdata")
```
before filtering small reads, separate and plot the controls and samples
```{r pheatmap}
rm(list=ls())
load("physeq_spongetank_20211220.rdata")
OTU<- data.frame(t(as.data.frame(otu_table(physeq_spongetank))))
s.reads<-sort(apply(OTU,1,sum))#13331.82, 6192.966
sd(s.reads);mean(s.reads)
o.reads<-sort(apply(OTU,2,sum))#29814.04, 10021.35
sd(o.reads); mean(o.reads)

metadata<- data.frame(sample_data(physeq_spongetank))
OTU.sam=OTU %>% filter(metadata$hour!="env" & metadata$hour!="CONTROL" & metadata$hour!="Pos")
OTU.ctl=OTU %>% filter(metadata$hour=="env" | metadata$hour=="CONTROL" | metadata$hour=="Pos")
metadata.ctl=metadata %>% filter(hour=="env" | hour=="CONTROL" | hour=="Pos")
metadata.sam=metadata %>% filter(hour!="env" & hour!="CONTROL" & hour!="Pos")

#set group
communityphm<-OTU.ctl
communityphm$control<-metadata.ctl$tank
communityphm<-arrange(communityphm, control)
metadata.ctl<-arrange(metadata.ctl, tank)
communityphm <- communityphm %>% dplyr::select(-control)
communityphm2=communityphm
communityphm2[communityphm2 <= 10] <- 0

communityphm=log(communityphm)
communityphm[communityphm == "-Inf"] <- 0
exclude = which(colSums(communityphm) == 0)
communityphm=communityphm[,-exclude]
or<-order(names(communityphm))
or<-c(1,4,9,3,5,11,8,2,26,10,6,7,12:25)
communityphm<-communityphm[,or]

annotation_row = data.frame(control=metadata.ctl$tank)
rownames(annotation_row) = metadata.ctl$name
ann_colors = list(tank = c(	FILT= "#E41A1C",CONTROL= "#377EB8",	EXT="#4DAF4A",PCR="#984EA3",SW="#FF7F00",TANKA="#FFFF33",TANKB="#A65628",TANKC="#F781BF"))
colvec <- brewer.pal(9, "Blues")
colvec <-c("white",colvec,"#053061") 
#ann_colors<- brewer.pal(8, "Set1")
pheatmap(communityphm,annotation_colors = ann_colors,annotation_row = annotation_row,cluster_rows = F,cluster_cols = F,border_color = NA,color= colvec, clustering_method="ward.D2",fontsize_row=8,margins=c(5,10),gaps_row = c(16),gaps_col = c(11))
#remove small reads and plot again
communityphm2=log(communityphm2)
communityphm2[communityphm2 == "-Inf"] <- 0
exclude2 = which(colSums(communityphm2) == 0)
communityphm2=communityphm2[,-exclude2]
or<-order(names(communityphm2))
or<-c(1,3,4,6,20,2,5,7:19)
communityphm2<-communityphm2[,or]
pheatmap(communityphm2,annotation_colors = ann_colors,annotation_row = annotation_row,cluster_rows = F,cluster_cols = F,border_color = NA,color= colvec, clustering_method="ward.D2",fontsize_row=8,margins=c(5,10))
rm(communityphm,communityphm2)

#set group
communityphm<-OTU.sam
communityphm$control<-metadata.sam$tank
communityphm<-arrange(communityphm, control)
metadata.sam<-arrange(metadata.sam, tank)
communityphm <- communityphm %>% dplyr::select(-control)
communityphm2=communityphm
communityphm2[communityphm2 <= 10] <- 0

communityphm=log(communityphm)
communityphm[communityphm == "-Inf"] <- 0
exclude = which(colSums(communityphm) == 0)
communityphm=communityphm[,-exclude]
or<-order(names(communityphm))
or<-c(1,4,9,3,5,11,8,2,26,10,6,7,12:25)
communityphm<-communityphm[,or]

annotation_row = data.frame(control=metadata.sam$tank)
rownames(annotation_row) = metadata.sam$name
ann_colors = list(tank = c(TANKA="#FFFF33",TANKB="#A65628",TANKC="#F781BF"))
colvec <- brewer.pal(9, "Blues")
colvec <-c("white",colvec,"#053061") 
#ann_colors<- brewer.pal(8, "Set1")
pheatmap(communityphm,annotation_colors = ann_colors,annotation_row = annotation_row,cluster_rows = F,cluster_cols = F,border_color = NA,color= colvec, clustering_method="ward.D2",fontsize_row=8,margins=c(5,10))
#,gaps_row = c(16),gaps_col = c(11)
#remove small reads and plot again
communityphm2=log(communityphm2)
communityphm2[communityphm2 == "-Inf"] <- 0
exclude2 = which(colSums(communityphm2) == 0)
communityphm2=communityphm2[,-exclude2]
or<-order(names(communityphm2))
or<-c(1,3,4,6,20,2,5,7:19)
communityphm2<-communityphm2[,or]
pheatmap(communityphm2,annotation_colors = ann_colors,annotation_row = annotation_row,cluster_rows = F,cluster_cols = F,border_color = NA,color= colvec, clustering_method="ward.D2",fontsize_row=8,margins=c(5,10))
#rm(communityphm)

```

now filter reads<10 and summary the reads in assigned/unassigned OTU, write them into a table 
```{r}
rm(list=ls())
load("physeq_spongetank_20211220.rdata")
OTU<- data.frame(t(as.data.frame(otu_table(physeq_spongetank))))
metadata<- data.frame(sample_data(physeq_spongetank))
OTU[OTU <= 10] <- 0
#summary the total number of reads
summary1=data.frame(name=names(OTU),sum=colSums(OTU))
#from now on, we only analysis samples and exclude controls
OTU.sam=OTU %>% filter(metadata$hour!="env" & metadata$hour!="CONTROL" & metadata$hour!="Pos")
exclude = which(colSums(OTU.sam) == 0)
OTU.sam=OTU.sam[,-exclude]

#plot the species compostion of each sample, group species into unassigned, target and 
OTU.pro=OTU.sam
names(OTU.pro)
OTU.pro$unassigned<-apply(OTU.pro[,c(13:44)],1,sum,na.rm=F)
OTU.pro <- OTU.pro %>% select(-c(OTU18:OTU57))
names(OTU.pro)
OTU.pro$extraneous<-apply(OTU.pro[,c(3:7,9,10,12)],1,sum,na.rm=F)
OTU.pro <- OTU.pro %>% select(-c(Pangasianodon_hypophthalmus,Brevoortia_tyrannus,Siganus_sp,f_Phasianidaesp,Mallotus_villosus,Halichoeres_sp,Pomadasys_sp,f_Astropectinidaesp))
OTU.pro$targeted<-apply(OTU.pro[,c(2:6)],1,sum,na.rm=F)
OTU.pro <- OTU.pro %>% select(-c(Amphiprion_ocellaris:Zebrasoma_veliferum))
OTU.pro=rotate_df(OTU.pro)
rownames=rownames(OTU.pro)
OTU.pro<- apply(OTU.pro,2,as.numeric)
OTU.pro=data.frame(apply(OTU.pro, 2, function(x) x/sum(x)))
rownames(OTU.pro)=rownames
OTU.pro[OTU.pro == "NaN"] <- 0
OTU.pro=rotate_df(OTU.pro)
metadata.sam=metadata %>% filter(hour!="env" & hour!="CONTROL" & hour!="Pos")

vector=c("SP.CREAM","SP.PINK","WATER")
vector2=c("TANKA","TANKB","TANKC")
OTU.pro.plot=data.frame()
OTU.pro.plot2=data.frame()
for (i in vector) {
  temp1=OTU.pro %>% filter(metadata.sam$sponge_type==i)
  metadata.temp=metadata.sam %>% filter(metadata.sam$sponge_type==i)
  for (j in vector2){
temp=temp1 %>% filter(metadata.temp$tank==j)
temp=as.matrix(temp)
temp <- reshape2::melt(temp)
names(temp) <- c("sample","species","proportion")
temp$tank=j
OTU.pro.plot=bind_rows(OTU.pro.plot,temp)
  }
  rm(temp,temp1,metadata.temp)
OTU.pro.plot$sample_type=i
OTU.pro.plot2=bind_rows(OTU.pro.plot2,OTU.pro.plot)
OTU.pro.plot=data.frame()
}
rm(OTU.pro.plot)

plot=function(x){
ggplot(x,mapping = aes(sample,proportion,fill=species))+geom_bar(stat='identity',position='fill') +labs(x = 'Sample',y = 'proportion') +scale_x_discrete(labels =c("fish","20h","24h","28h","52h","76h","100h"))+theme_classic()
}

OTU.pro.plot3=OTU.pro.plot2 %>% filter(sample_type=="WATER")
OTU.pro.plotA=OTU.pro.plot3 %>% filter(tank=="TANKA")
OTU.pro.plotB=OTU.pro.plot3 %>% filter(tank=="TANKB")
OTU.pro.plotC=OTU.pro.plot3 %>% filter(tank=="TANKC")

p1=plot(OTU.pro.plotA)
p2=plot(OTU.pro.plotB)
p3=plot(OTU.pro.plotC)
ggpubr::ggarrange(p1,p2,p3,ncol=3,nrow=1,labels=c("A","B","C"),common.legend = T)
```

#######third filter: here we only keep targeted species for following analysis#####
filtering OTU that show in "day one sponge samples", which are carried into the tanks by rocks and spongeself, one OTU (contamination) show in sterivex filter blank, and all unassigned OTUs.
```{r third filter}
#remove positive control and human
OTU.sam=OTU.sam %>% select(Amphiprion_ocellaris,Chromis_viridis,Zebrasoma_veliferum,Hippocampus_semispinosis,Gramma_loreto)
TAX=data.frame(tax_table(physeq_spongetank))
OTU.sam$name=rownames(OTU.sam)
#i remove the contamination by hand, cause apparently the sponge brought it in from outside.
r1=which(OTU.sam$name == "D2.TANKB.FISH.SPONGE.PINK")
r2=which(OTU.sam$name == "D2.TANKC.FISH.SPONGE.CREAM")
OTU.sam$Zebrasoma_veliferum[r1]="0"
OTU.sam$Zebrasoma_veliferum[r2]="0"
OTU.sam$Zebrasoma_veliferum=as.numeric(OTU.sam$Zebrasoma_veliferum)
OTU.sam=OTU.sam %>% select(-name)
metadata.sam
OTU.sam=OTU.sam %>% filter(metadata.sam$sponge_type !="SP.BLACK")
metadata.sam=metadata.sam %>% filter(sponge_type !="SP.BLACK")
save(OTU.sam,metadata.sam,TAX,file = "sample_data_only_20211220.rdata")
```

#######statistical analysis start here##########
Q1:Does reads relate to detection rate or species size?
```{r}
rm(list=ls())
load("sample_data_only_20211220.rdata")
OTU.sam=log(OTU.sam)
OTU.sam[OTU.sam == "-Inf"] <- 0
OTU.sam2=OTU.sam
OTU.sam2[OTU.sam2 > 0] <- 1
speciesinfo.t=data.frame(detection=colSums(OTU.sam2),logreads=colSums(OTU.sam),
name=c("Amphiprion ocellaris","Chromis viridis","Zebrasoma veliferum","Hippocampus semispinosis","Gramma loreto"))

ggplot(data = speciesinfo.t,aes(x = detection, y = logreads,label=name))+geom_point()+theme_classic()+ylab("log(reads)")+geom_smooth(method = "lm",se=T)+geom_text_repel(size=2)+scale_x_continuous(limits=c(0,35),breaks=seq(5,35,5))

#seprate sample type,c("SP.CREAM","SP.PINK","WATER")
OTU.sam.p=OTU.sam %>% filter(metadata.sam$sponge_type=="SP.PINK")
OTU.sam.p2=OTU.sam.p
OTU.sam.p2[OTU.sam.p2 > 0] <- 1
speciesinfo.p=data.frame(detection=colSums(OTU.sam.p2),logreads=colSums(OTU.sam.p),
name=c("Amphiprion ocellaris","Chromis viridis","Zebrasoma veliferum","Hippocampus semispinosis","Gramma loreto"))
speciesinfo.p$sample="PINK"
OTU.sam.c=OTU.sam %>% filter(metadata.sam$sponge_type=="SP.CREAM")
OTU.sam.c2=OTU.sam.c
OTU.sam.c2[OTU.sam.c2 > 0] <- 1
speciesinfo.c=data.frame(detection=colSums(OTU.sam.c2),logreads=colSums(OTU.sam.c),
name=c("Amphiprion ocellaris","Chromis viridis","Zebrasoma veliferum","Hippocampus semispinosis","Gramma loreto"))
speciesinfo.c$sample="CREAM"
OTU.sam.w=OTU.sam %>% filter(metadata.sam$sponge_type=="WATER")
OTU.sam.w2=OTU.sam.w
OTU.sam.w2[OTU.sam.w2 > 0] <- 1
speciesinfo.w=data.frame(detection=colSums(OTU.sam.w2),logreads=colSums(OTU.sam.w),
name=c("Amphiprion ocellaris","Chromis viridis","Zebrasoma veliferum","Hippocampus semispinosis","Gramma loreto"))
speciesinfo.w$sample="WATER"
speciesinfo=bind_rows(speciesinfo.w,speciesinfo.p,speciesinfo.c)
col=c("#FFB90F", "#8B008B", "#7AC5CD")
ggplot(data = speciesinfo,aes(x = detection, y = logreads,color=sample,label=name,shape=name))+geom_point()+theme_classic()+ylab("log(reads)")+geom_text_repel(size=2)+scale_x_continuous(limits=c(0,10),breaks=seq(0,10,1))+scale_color_manual(values=col)
```
A: for all, Adjusted R-squared:  0.9727 p= 0.00125 ** high relation

Q2:Does reads and detection rate relate to time, differ from type?
```{r}
#total
metadata.sam$sampleread=rowSums(OTU.sam)
metadata.sam$hour=as.numeric(as.character(metadata.sam$hour))
ggplot(data = metadata.sam,aes(x = hour, y = sampleread,color=sponge_type))+geom_point()+theme_classic()+geom_smooth(method = "glm",se=F)
summary(lm(metadata.sam$hour~metadata.sam$sampleread))
#pink
metadata.sam.p=metadata.sam %>% filter(sponge_type=="SP.PINK")
summary(lm(metadata.sam.p$hour~metadata.sam.p$sampleread))
#cream
metadata.sam.c=metadata.sam %>% filter(sponge_type=="SP.CREAM")
summary(lm(metadata.sam.c$hour~metadata.sam.c$sampleread))
#water
metadata.sam.w=metadata.sam %>% filter(sponge_type=="WATER")
summary(lm(metadata.sam.w$hour~metadata.sam.w$sampleread))
plot(TukeyHSD(aov(metadata.sam$sampleread~metadata.sam$sponge_type)))
```
A:cream and water total reads decrease along time, but pink not.

Q3:what is the pattern of each sponge and water according to species richness and community
now look at Î± diversity
```{r incidence lm model}
OTU.sam2<- data.frame(apply(OTU.sam2,2,as.numeric))
OTU.sam2$incidence=rowSums(OTU.sam2)
OTU.full.i=bind_cols(metadata.sam,OTU.sam2)
OTU.full.i=OTU.full.i %>% filter(sponge_type =="WATER" | sponge_type =="SP.PINK" | sponge_type =="SP.CREAM")
sample_data<-read.csv("sampledata_full_20211207.csv", header=TRUE)
#sample_data=arrange(sample_data,name)
sample_data=left_join(sample_data,OTU.full.i,by="name")
rownames(sample_data)=sample_data$name
sample_data=sample_data%>%filter(sponge_type.x=="SP.PINK" | sponge_type.x=="SP.BLACK" |sponge_type.x=="SP.CREAM" | sponge_type.x=="WATER")
sample_data[is.na(sample_data)]<-0
OTU.full.i=sample_data%>%select(tank.x:sponge_type.x,Amphiprion_ocellaris:incidence)
names(OTU.full.i)[1:6]=c("tank","hour","hours_fish","date","time","sponge_type")

OTU.full.i <- OTU.full.i %>%
  mutate(
    hours_fish = case_when(
      hours_fish == "0" ~ "fish",
      hours_fish == "20" ~ "20h",
      hours_fish == "24" ~ "24h",
      hours_fish == "28" ~ "28h",
      hours_fish == "44" ~ "52h",
      hours_fish == "68" ~ "76h",
      hours_fish == "92" ~ "100h",
      TRUE ~ as.character(hours_fish)
    )
  )
OTU.full.i <- OTU.full.i %>%
  mutate(
    hour = case_when(
      hour == "0" ~ "0",
      hour == "20" ~ "20",
      hour == "24" ~ "24",
      hour == "28" ~ "28",
      hour == "44" ~ "52",
      hour == "68" ~ "76",
      hour == "92" ~ "100",
      TRUE ~ as.character(hour)
    )
  )
OTU.full.i <- OTU.full.i %>%
  mutate(
    sponge_type = case_when(
      sponge_type == "SP.BLACK" ~ "BLACK",
      sponge_type == "SP.PINK" ~ "PINK",
      sponge_type == "SP.CREAM" ~ "CREAM",
      TRUE ~ as.character(sponge_type)
    )
  )
OTU.full.i$sponge_type=as.factor(OTU.full.i$sponge_type)

OTU.full.i$hours_fish=as.factor(OTU.full.i$hours_fish)
OTU.full.i$hour=as.numeric(OTU.full.i$hour)

OTU.full.i$hours_fish=factor(OTU.full.i$hours_fish,levels = c('fish','20h','24h','28h','52h','76h','100h')) 
OTU.full.i$sponge_type=as.factor(as.character(OTU.full.i$sponge_type))
OTU.full.i$tank=as.factor(as.character(OTU.full.i$tank))

ggplot(OTU.full.i, aes(x=factor(hours_fish), y=incidence,fill=sponge_type)) + 
  geom_boxplot() +
  geom_jitter(width =0.1,size=1,height=0)+ 
  theme_bw()+ylab("Observed species richness")+xlab("Time (hour)")+scale_fill_manual(values=c("#8B7D6B", "#FFB90F", "#8B008B","#7AC5CD"))+theme_bw() + 
  theme(axis.ticks = element_blank()) +theme(panel.grid=element_blank())+ theme(legend.title=element_blank()) #+geom_smooth(method = "loess", se=FALSE, color="black", aes(group=sponge_type))

p2=ggplot(OTU.full.i, aes(x = hour, y = incidence, color=sponge_type)) +geom_jitter(aes(colour=sponge_type),height=0.0,width =0.8,size=1.5)+geom_smooth(method = "lm",se=F)+ylab("Observed species richness")+xlab("Time (hour)")+scale_color_manual(values=c("#8B7D6B", "#FFB90F", "#8B008B","#7AC5CD"))+theme_bw()+theme(panel.grid=element_blank())+theme(legend.title=element_blank())+scale_x_continuous( limits=c(0,100),breaks=c(0,20,24,28,52,76,100))+scale_y_continuous( limits=c(0,3),breaks=c(0,1,2,3))
ggpubr::ggarrange(p1,p2,ncol=2,nrow=1,labels=c("A","B"),common.legend = T)

#LM
OTU.full.i.WATER=OTU.full.i%>%filter(sponge_type=="WATER")
lm.WATER<-lm(OTU.full.i.WATER$incidence~OTU.full.i.WATER$hour)
summary(lm.WATER)# p-value =0.000737 ***,r2=0.4308

OTU.full.i.PINK=OTU.full.i%>%filter(sponge_type=="PINK")
lm.pink<-lm(OTU.full.i.PINK$hour~OTU.full.i.PINK$incidence)
summary(lm.pink)# p-value =0.1403,r2=0.07186

OTU.full.i.CREAM=OTU.full.i%>%filter(sponge_type=="CREAM")
lm.CREAM<-lm(OTU.full.i.CREAM$hour~OTU.full.i.CREAM$incidence)
summary(lm.CREAM)# p-value =0.005975**,r2=0.3
```

```{r incidence aov}
fit1<-aov(incidence~sponge_type,data=OTU.full.i)
summary(fit1)
#sponge_type  3  17.65   5.885   7.471 0.000182 ***
#remove black sponge
OTU.full.i2=OTU.full.i%>%filter(sponge_type!="SP.BLACK")
OTU.full.i2$sponge_type=as.factor(as.character(OTU.full.i2$sponge_type))
leveneTest(incidence~sponge_type,data=OTU.full.i2)# 0.1842
fit2<-aov(incidence~sponge_type,data=OTU.full.i2)
summary(fit2)
#             Df  Sum Sq Mean Sq F value Pr(>F)
#sponge_type  2   0.89  0.4444   0.346  0.709 
plot(fit2, 2)

fit3<-aov(incidence~sponge_type+hours_fish,data=OTU.full.i2)
plot(fit3, 2)
summary(fit3)
#sponge_type  2   0.89   0.444   0.470 0.627803    
#hours_fish   6  25.94   4.323   4.567 0.000815 ***

par(mar=c(4,4,2,2))
interaction2wt(incidence~sponge_type*hours_fish,data=OTU.full.i2)

TukeyHSD(aov(incidence~sponge_type,data=OTU.full.i2)) 
#plot(TukeyHSD(aov(incidence~sponge_type,data=OTU.full.i)))
# library(ggsignif)
# library(gtools)
# compared_list=list()
# levels=levels(as.factor(OTU.full3$sponge_type))
# combinations=combinations(4,2,levels)
# for (i in 1:(length(combinations)/2)) {
#   compared_list[[i]]=combinations[i,]
# }
# ggplot(OTU.full3, aes(x=sponge_type, y=incidence)) + 
#   geom_boxplot() +
#   labs(title = "incidence t.test")+
#   geom_signif(comparisons = compared_list, test = t.test, step_increase = 0.2,map_signif_level=TRUE)+geom_jitter(width =0.3,size=1,height=0.0)
```
the beta diversity analysis base on no black data
```{r mvabund}
OTU.sam2=OTU.sam2 %>%select(-incidence)
communityB.mvb <- mvabund(OTU.sam2)  
nboot <- 999
#base model1 for sponge_type
modt1.mvb <- manyglm(communityB.mvb~sponge_type, family="negative.binomial",data=metadata.sam)
plot(modt1.mvb)
anova(modt1.mvb,cor.type = "shrink", test = "score", show.time = "all", nBoot = nboot)
#                   Res.Df   Df.diff   score      Pr(>score)
#sponge_type     56       4            0.001 ***
modt2.mvb <- manyglm(communityB.mvb~tank, family="binomial",data=metadata.sam)
plot(modt2.mvb)
anova(modt2.mvb,cor.type = "shrink", test = "score", show.time = "all", nBoot = nboot)
#tank            56       4            0.001 ***

communityB.mvb <- mvabund(OTU.sam.c2) 
modt3.mvb <- manyglm(communityB.mvb~hour*tank, family="binomial",data=metadata.sam.c)
plot(modt3.mvb)
anova(modt3.mvb,cor.type = "shrink", test = "score", show.time = "all", nBoot = nboot)
#hour            19       1 10.44      0.021 *  
#tank            15       4            0.001 ***
#hour:tank       15       3            0.001 ***
communityB.mvb <- mvabund(OTU.sam.p2) 
modt4.mvb <- manyglm(communityB.mvb~hour*tank, family="binomial",data=metadata.sam.p)
plot(modt4.mvb)
anova(modt4.mvb,cor.type = "shrink", test = "score", show.time = "all", nBoot = nboot)
#hour            17       1 2.846      0.529    
#tank            13       4            0.001 ***
#hour:tank       13       3            0.001 ***
communityB.mvb <- mvabund(OTU.sam.w2) 
modt5.mvb <- manyglm(communityB.mvb~hour*tank, family="binomial",data=metadata.sam.w)
plot(modt5.mvb)
anova(modt5.mvb,cor.type = "shrink", test = "score", show.time = "all", nBoot = nboot)
#hour            17       1 13.46      0.007 ** 
#tank            13       4            0.001 ***
#hour:tank       13       3            0.001 ***
```
plot pyramid: community change along time
```{r Pyramid}
brewer.pal(n =11, name = "RdYlBu")
col1=c("#E41A1C", "#FB8072", "#377EB8" )
col2=c("#E41A1C", "#FB8072", "#80B1D3" )
col3=c("#E41A1C", "#FB8072", "#ABD9E9" )

OTU.full=bind_cols(metadata.sam,OTU.sam)
OTU.full=OTU.full %>% filter(sponge_type =="WATER" | sponge_type =="SP.PINK" | sponge_type =="SP.CREAM")
sample_data<-read.csv("sampledata_full_20211207.csv", header=TRUE)
#sample_data=arrange(sample_data,name)
sample_data=left_join(sample_data,OTU.full,by="name")
rownames(sample_data)=sample_data$name
sample_data=sample_data%>%filter(sponge_type.x=="SP.PINK" | sponge_type.x=="SP.BLACK" |sponge_type.x=="SP.CREAM" | sponge_type.x=="WATER")
sample_data[is.na(sample_data)] <- 0
OTU.full=sample_data%>%select(tank.x:sponge_type.x,Amphiprion_ocellaris:Gramma_loreto)
names(OTU.full)[1:6]=c("tank","hour","hours_fish","date","time","sponge_type")

OTU.full2=OTU.full %>% filter(OTU.full$tank=="TANKB" & OTU.full$sponge_type =="SP.CREAM")
rowname=OTU.full2$name
OTU.full3=OTU.full2 %>% select(Amphiprion_ocellaris:Gramma_loreto)
exclude2 = which(colSums(OTU.full3) == 0)
OTU.full3=OTU.full3[,-exclude2]

OTU.full3=as.matrix(OTU.full3)
OTU.full3 <- reshape2::melt(OTU.full3)
names(OTU.full3) <- c("sample","species","reads")

ggplot(OTU.full3, aes(x =sample , y = reads, fill = species)) +
  geom_bar(stat = "identity", position='stack') +  
  theme_bw() + 
  theme(axis.ticks = element_blank()) + 
  theme(axis.text.x = element_text(angle=90, hjust=1, vjust=.5))+theme(panel.grid=element_blank()) +    
  theme(panel.border=element_blank())  + 
  theme(legend.title = element_blank(), legend.position = "left") + 
 scale_fill_manual(values=col2)+scale_y_continuous(limits=c(0,30),breaks=seq(0,30,5))
```

